### PR TITLE
add description field to service_account for Terraform

### PR DESCRIPTION
**Pull request by tmshn**
I've added the `description` field to resource `google_service_account`

ref:
- https://godoc.org/google.golang.org/api/iam/v1#ProjectsServiceAccountsService.Patch
- https://cloud.google.com/iam/reference/rest/v1/projects.serviceAccounts#ServiceAccount

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
`The length of the `description` field of `google_service_account` is now limited to 256 characters.
```

### CODE REVIEW

**Comment by tmshn** [comment]
@ndmckinley Should I close this PR due to #2513 ?

**Comment by ndmckinley** [comment]
Yours is definitely the better version of this PR.  I'll let @tysen decide what to do - maybe we should undo that PR and merge this one instead?

**Comment by tmshn** [comment]
@ndmckinley As another choice; I'm happy to merge master and resolve conflicts between our codes ( `time.Sleep(time.Second)` is something missing in my code).

**Comment by tysen** [comment]
>As another choice; I'm happy to merge master and resolve conflicts between our codes ( `time.Sleep(time.Second)` is something missing in my code).

@tmshn - That sounds like a good plan to me.

**Comment by tmshn** [comment]
@tysen resolved 3b1671c

**Comment by tysen** [comment]
Looks like there's some errors in the build.

**Comment by tmshn** [comment]
> Looks like there's some errors in the build.

@tysen Sorry I didn't notice that! That's just a typo, now fixed: 1798ab1

**Comment by tmshn** [comment]
And I solved the conflict with #2650 by @ndmckinley (actually his new logic is almost equivalent to mine, so the diff against master is getting smaller).

File: third_party/terraform/resources/resource_google_service_account.go [hunk 1]
```diff
@@ -121,31 +123,33 @@ func resourceGoogleServiceAccountDelete(d *schema.ResourceData, meta interface{}
...
-			UpdateMask: strings.Join(updateMask, ","),
-			ServiceAccount: &iam.ServiceAccount{
-				DisplayName: d.Get("display_name").(string),
-				Description: d.Get("description").(string),
-				Etag:        sa.Etag,
-			},
-		}).Do()
-	if err != nil {
-		return err
+	if len(updateMask) > 0 {
```
Comment (tysen) [inline third_party/terraform/resources/resource_google_service_account.go]: What's the motivation behind these changes to `..Update`? I'm not sure the Terraform SDK would call into `..Update` without a diff.

Comment (tmshn) [reply third_party/terraform/resources/resource_google_service_account.go]: Ah... this happened during merging #2650 by @ndmckinley into my original code 81c263d94bc541c8b01a8fe32ce97dcbdd8e376a, but well you are right, we don't need this.

File: third_party/terraform/tests/resource_google_service_account_test.go [hunk 2]
```diff
@@ -18,7 +18,7 @@ func TestAccServiceAccount_basic(t *testing.T) {
 	displayName := "Terraform Test"
 	displayName2 := "Terraform Test Update"
 	desc := "test description"
-	desc2 := "test description update"
+	desc2 := ""
```
Comment (tmshn) [inline third_party/terraform/tests/resource_google_service_account_test.go]: This diff might also seem to be needless, but IMO setting empty value tend to be more buggy than setting other values, so I believe this is meaningful.

File: third_party/terraform/resources/resource_google_service_account.go (hunk 1 above)
Comment (tmshn) [reply third_party/terraform/resources/resource_google_service_account.go]: fixed ea0839f

**Comment by tysen** [comment]
Merging failed - @tmshn do you mind rebasing?

**Comment by tmshn** [comment]
@tysen 

> This validation isn't strictly correct - the docs say "Must be less than or equal to 256 UTF-8 bytes" (bytes, not characters)

`len` against `string` in Golang returns bytes count, which means `valiation.StringLenBetween` validates based on bytes count rather than characters count. So I think the validation is correct, isn't it? (maybe I should change the text for release note?)

> Merging failed - @tmshn do you mind rebasing?

Rebased.
(I should have abandoned this PR earlier and open new one only to add validation...sorry to make the situation complicated.)
